src/components/GroupAndSearchDropdown/index.js

```diff
@@ -106,47 +118,50 @@ export const GroupAndSearchDropdown = ({
-                }}
-                className="contacto-icon--input-suffix-variable-dropdown"
-              >
-                <Icon svg={dropdownIcon} size={20} />
-              </div>
-            }
-            onClick={(e) => !openOnTextboxClick && e.stopPropagation()}
-            onKeyUp={(e) => {
-              e.persist()
-              const value = e.target.value
-              // const keyPressed = String.fromCharCode(e.keyCode)
-              const caretEnd = e.target.selectionEnd
-              const keyPressed = value[caretEnd - 1]
-              const openDropdown =
-                value.slice(-2) === '{{' ||
-                (keyPressed === '|' && /\{\{[\w|]+\|$/g.test(value.substring(0, caretEnd)))
-              setShowDropdown(openDropdown)
-            }}
-            onChange={(e) => {
-              // So that the event is not lost
-              e.persist()
-              const caretStart = e.target.selectionStart
-              const caretEnd = e.target.selectionEnd
-              onChange(e)
+        {onlyDropdown ? (
+          <div></div>
+        ) : (
+          <div className={`group-dropdown`}>
+            <TextField
+              type={'text'}
+              ref={textFieldRef}
+              suffix={
+                <div
+                  onClick={(e) => {
+                    e.stopPropagation()
+                    setShowDropdown(!showDropdown)
+                  }}
+                  className="contacto-icon--input-suffix-variable-dropdown"
+                >
+                  <Icon svg={dropdownIcon} size={20} />
+                </div>
+              }
+              onClick={(e) => !openOnTextboxClick && e.stopPropagation()}
+              onKeyUp={(e) => {
+                e.persist()
+                const value = e.target.value
+                // const keyPressed = String.fromCharCode(e.keyCode)
+                const caretEnd = e.target.selectionEnd
+                const keyPressed = value[caretEnd - 1]
+                const openDropdown =
+                  value.slice(-2) === '{{' ||
+                  (keyPressed === '|' && /\{\{[\w|]+\|$/g.test(value.substring(0, caretEnd)))
+                setShowDropdown(openDropdown)
+              }}
+              onChange={(e) => {
+                // So that the event is not lost
+                e.persist()
+                const caretStart = e.target.selectionStart
+                const caretEnd = e.target.selectionEnd
+                onChange(e)
 
-              // To preserve the caret position
-              setTimeout(() => e.target.setSelectionRange?.(caretStart, caretEnd), 0)
-            }}
-            {...props}
-          />
-        </div>
+                // To preserve the caret position
+                setTimeout(() => e.target.setSelectionRange?.(caretStart, caretEnd), 0)
+              }}
+              {...props}
+            />
+          </div>
+        )}
       </Dropdown>
     </>
   )
```

Comment:
**Suggestion:** Extract the logic for handling the dropdown and text field into a separate function to improve readability and maintainability.
```suggestion
        {renderTextField()}

        // Add this function above the return statement
        function renderTextField() {
          if (onlyDropdown) {
            return <div></div>;
          }

          const handleKeyUp = (e) => {
            e.persist();
            const value = e.target.value;
            const caretEnd = e.target.selectionEnd;
            const keyPressed = value[caretEnd - 1];
            const openDropdown =
              value.slice(-2) === '{{' ||
              (keyPressed === '|' && /\{\{[\w|]+\|$/g.test(value.substring(0, caretEnd)));
            setShowDropdown(openDropdown);
          };

          const handleChange = (e) => {
            e.persist();
            const caretStart = e.target.selectionStart;
            const caretEnd = e.target.selectionEnd;
            onChange(e);
            setTimeout(() => e.target.setSelectionRange?.(caretStart, caretEnd), 0);
          };

          return (
            <div className={`group-dropdown`}>
              <TextField
                type={'text'}
                ref={textFieldRef}
                suffix={
                  <div
                    onClick={(e) => {
                      e.stopPropagation();
                      setShowDropdown(!showDropdown);
                    }}
                    className="contacto-icon--input-suffix-variable-dropdown"
                  >
                    <Icon svg={dropdownIcon} size={20} />
                  </div>
                }
                onClick={(e) => !openOnTextboxClick && e.stopPropagation()}
                onKeyUp={handleKeyUp}
                onChange={handleChange}
                {...props}
              />
            </div>
          );
        }
```